alert-common/src/main/java/com/synopsys/integration/alert/common/SetMap.java

```diff
@@ -0,0 +1,54 @@
+package com.synopsys.integration.alert.common;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class SetMap<K, S> {
+    private final Map<K, Set<S>> map;
+
+    public SetMap(final Map<K, Set<S>> map) {
+        this.map = map;
+    }
```

Comment:
Can we also create a default constructor that doesn't require a map to be passed and just starts empty?

Reply:
Why not make this class `extend Map<K,Set<S>>`? This way it conforms to a Map.